src/shared/ui/components/TransactionsTable.vue

```diff
@@ -0,0 +1,367 @@
+<script setup lang="ts">
+import { computed, ref, watch } from 'vue';
+import { useTable } from '@/shared/lib/table';
+import * as http from '@/shared/api';
+import { objectOmit } from '@vueuse/shared';
+import {
+  type filterTransactionsModel as ftm,
+  TransactionStatusFilter,
+  TransactionTypeFilter,
+} from '@/features/filter-transactions';
+import type { AccountId } from '@/shared/api/schemas';
+import { useErrorHandlers } from '@/shared/ui/composables/useErrorHandlers';
+import { accountDetailsAdaptiveOptions, defaultAdaptiveOptions } from '@/features/filter-transactions/adaptive-options';
+import { defaultFormat } from '@/shared/lib/time';
+import TransactionStatus from '@/entities/transaction/TransactionStatus.vue';
+import BaseTable from '@/shared/ui/components/BaseTable.vue';
+import BaseHash from '@/shared/ui/components/BaseHash.vue';
+import BaseLink from '@/shared/ui/components/BaseLink.vue';
+import { useWindowSize } from '@vueuse/core';
+
+const { handleUnknownError } = useErrorHandlers();
+const props = withDefaults(
+  defineProps<{
+    defaultTabs?: boolean
+    authority?: AccountId | null
+    block?: number
```

Comment:
```suggestion
    filterBy: { kind: 'authority', id: AccountId } | { kind: 'block', block: number }
```

Stronger types, clearer usage, less chance to make a mistake